ceph_deploy/hosts/remotes.py

```diff
@@ -45,6 +49,25 @@ def platform_information(_linux_distribution=None):
     )
 
 
+def parse_os_release(release_path='/etc/os-release'):
+    """ Extract (distro, release, codename) from /etc/os-release if present """
+    release_info = {}
+    if os.path.isfile(release_path):
+        for line in open(release_path, 'r').readlines():
+            line = line.strip()
+            if line.startswith('#'):
+                continue
+            parts = line.split('=')
+            if len(parts) != 2:
+                continue
+            release_info[parts[0].strip()] = parts[1].strip("\"'\n\t ")
+    # The below should be about right, per
+    # https://www.freedesktop.org/software/systemd/man/os-release.html,
+    distro = release_info.get('NAME', '')
+    release = release_info.get('VERSION_ID', '')
+    codename = release_info.get('VERSION_CODENAME', release_info.get('VERSION', ''))
```

Comment:
This would not work with Ubuntu (`codename` would return `''`):

    $ cat /etc/os-release
    NAME="Ubuntu"
    VERSION="16.04 LTS (Xenial Xerus)"
    ID=ubuntu
    ID_LIKE=debian
    PRETTY_NAME="Ubuntu 16.04 LTS"
    VERSION_ID="16.04"
    HOME_URL="http://www.ubuntu.com/"
    SUPPORT_URL="http://help.ubuntu.com/"
    BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
    UBUNTU_CODENAME=xenial

Reply:
It tries VERSION_CODENAME first, then falls back to VERSION, so in this case it'll return `codename == '16.04 LTS (Xenial Xerus)'`. But yeah, we'd want it to return `'xenial'` in this case.